### DISCUSSION TITLE
runDLKcat() cannot find path to data

### DISCUSSION THREAD

**Discussion by PkiwiBird** [SysBioChalmers/GECKO#378] [Q&A] [answered]
When runing `runDLKcat(); `there's this error:
```
Running DLKcat prediction, this may take many minutes, especially the first time.
docker: Error response from daemon: create ~/drive/Recon4IMD/WP2_MetNetwork/T2.5_Kinetics/data: "~/drive/Recon4IMD/WP2_MetNetwork/T2.5_Kinetics/data" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.
Error using runDLKcat
DLKcat encountered an error or it did not create any output file.
```
when running docker run --help gave error:
`Unrecognized function or variable 'docker'.`

From the terminal docker seems to be installed:
```
 which docker
/usr/bin/docker
```

**Comment by mihai-sysbio** [comment]
Hi @PkiwiBird! Could you try using the full path, starting with a `/`, instead of starting with the `~` alias?

**Reply by mihai-sysbio** [reply, marked as answer]
To give more context, `~` is your home directory, but I wouldn't be surprised if the new shell, where the Docker command is run, has not been passed this alias, so it does not understand it, as if it were a relative path `../.../drive/`, which would explain the error message.

**Reply by PkiwiBird** [reply]
Thank you. I don't like full paths. Do you know how to add that path to docker? just adding to .bashrc doesn't seem to work.

**Reply by mihai-sysbio** [reply]
Modifying the environment in which Docker runs would require altering the GECKO source code.

A quicker alternative would be to convert the path from avoid the `~` would be to convert it. This SO solution would work well https://stackoverflow.com/a/75470259 I think.

**Reply by PkiwiBird** [reply]
thanks!

**Reply by edkerk** [reply]
I assume that `params.path` in the model adapter started with ~? We can make `runDLKcat` turn it into a full path before calling docker, using the code @mihai-sysbio suggested.

**Reply by mihai-sysbio** [reply]
That's an interesting idea @edkerk. Modifying parameters such as paths would have to be carefully done (so that it works on all systems), and clearly documented, since otherwise it wouldn't be obvious to debug. Perhaps best to apply this only when calling Docker as you are saying? Another alternative would be to print a warning if the path starts with `~`.
